tests/zfs-tests/cmd/mmap_sync.c

```diff
@@ -73,8 +76,8 @@ main(int argc, char *argv[])
 	filepath[0] = '\0';
 	char *file = &filepath[0];
 
-	strcat(file, testdir);
-	strcat(file, "/msync_file");
+	strlcat(file, testdir, 512);
+	strlcat(file, "/msync_file", 512 - strlen(testdir));
```

Comment:
Let's just write with this `snprintf` or even `asprintf` to avoid the workaround.

Reply:
I like that idea. This also can be done in `contrib/pam_zfs_key/pam_zfs_key.c` to avoid 3 rather unwieldy `strlcat()` invocations and also would work in #13876.

Done.